view/frontend/web/js/private-data.js

```diff
@@ -0,0 +1,23 @@
+define([
+    'uiComponent',
+    'Magento_Customer/js/customer-data',
+    'jquery'
+], function (Component, customerData, $) {
+    'use strict';
+
+    return Component.extend({
+        /** @inheritdoc */
+        initialize: function () {
+            this._super();
+            this.metrilosection = customerData.get('metrilo-private-data');
+
+            if (this.metrilosection) {
+                this.metrilosection.subscribe(function () {
+                    $.each(this.metrilosection().events, function(key, val){
```

Comment:
Have you considered the naming? What are the `key` and `val`?

Reply:
Hmm, didn't think about it, but the key is a representation for index and value is the actual event. Will rename them to index -> event.

Reply:
What do you need the index for?

Reply:
I do not need it, just thought that this is the way to do foreach in jQuery. Examples are from this doc: https://api.jquery.com/jquery.each/

Reply:
In there it is mentioned that you can use the `this` keyword to access the object of the current iteration. 
Furthermore, why do you use `jquery` at all? You could do with a simple `forEach`.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach

Reply:
Removed the jQuery from the ko component and used pure js forEach function.